extension/src/popup/ducks/blockaid.ts

```diff
@@ -0,0 +1,92 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+
+import { ActionStatus, ErrorMessage } from "@shared/api/types";
+import { INDEXER_URL } from "@shared/constants/mercury";
+import { NetworkDetails } from "@shared/constants/stellar";
+import { isCustomNetwork } from "@shared/helpers/stellar";
+
+export const scanSite = createAsyncThunk<
```

Comment:
just curious - what do you see as the value of saving the blockaid result in redux? it seems like we only need this result in the one view

Reply:
yeah actually I wondered the same thing and I think this is true for the balance/history ducks as well. The reason I did it here is because it's the most common pattern in this app for fetching data and using a data fetching lifecycle(ActionStatus). I think the alternative would be to either remove the `scanSiteResults` key and assign it locally in the component or maybe do away with the duck altogether and use a helper + local loading states. 

The downside of the helper is that then later if we do need this call in another place, we're duplicating the loading logic but that seems fine to me.

Reply:
Yeah, I def see your point about the ActionStatus pattern. Maybe we should just use the helper/local loading state pattern for now and optimize to use redux if we find out later on we need this data in a few different places?

I don't think your approach is wrong or anything, I think redux just causes us to write a lot of lines code with all the slice/reducers stuff. Might be nicer to keep it a little simpler if there isn't a big advantage to using redux

Reply:
👍 refactored to a helper hook in b8f20f03c413cfb70e6c40bce1500ca860ac9f05